### ISSUE TITLE
Custom Testing Assertions

### ISSUE THREAD

**Issue by julian94** [usebruno/bruno#688]
The current Testing Assertions are good for simple checks, but as real life use cases will inevitably vary, it would be useful to be able to extend the standard set with custom assertions.

Ideally these assertions could be shared across a whole collection by putting it in a something like `shared.bru`. But a good start could be to put them into the request files.

### Example implementation from user view

```bru
script:assertions {
  assertion("isHelloWorldMessage", function(value) {
    expect(value).to.equal("Hello World!");
  });

  assertion("isTwiceAsMuchAs", function(value, comparison) {
    expect(value).to.equal(comparison * 2);
  });
}


assert {
  res.body: isHelloWorldMessage
  res.status.code: isTwiceAsMuchAs 100
}
```

**Comment by helloanoop** [comment]
@julian94 Yes, I think supporting customer assertions is a good idea. For now, you can use scripting like below to segregate the common logic.

#### Folder Structure
```bash
|-- folder
|   |-- request1.bru
|   |-- request2.bru
|-- bruno.json
|-- environments
|   |-- dev.bru
|   |-- qa.bru
|-- asserts.js
```

`asserts.js`
```javascript
const isHelloWorldMessage = (value) => {
  expect(value).to.equal("Hello World!");
}

const isTwiceAsMuchAs = (value, expected) => {
  expect(value).to.equal(expected * 2);
}

module.exports = {
  isHelloWorldMessage,
  isTwiceAsMuchAs
}
```

Inside your `tests` tab, you could do.
```javascript
const {
  isHelloWorldMessage,
} = require('./asserts');

test("should return hello world", function() {
  const data = res.getBody();
  isHelloWorldMessage(data);
});
```

![image](https://github.com/usebruno/bruno/assets/9350733/eb7eb48e-9c42-45c1-9aee-997962c6f3a3)

**Comment by gh47110815** [comment]
.... and because developing assertions and scripts sometimes is an iterative process it would also be nice, if you could somehow run scripts and these global assertions against the last received response instead of sending a request on every change in a script or assertion. It would make live much more easier if there would be a chance to "rerun script" or "retest assertions" without the need of a real request-response situation, because you could concentrate on the assertion or the script and not on a maybe changing request (especially if the request does not only read data in the backend but also modifies data ;-)
